server_environment/models/server_env_mixin.py

```diff
@@ -0,0 +1,405 @@
+        * ``no_default_field``: disable creation of a field for storing
+          the default value, must be used with ``compute_default`` and
+          ``inverse_default``
+        * ``compute_default``: name of a compute method to get the default
+          value when no key is present in configuration files
+        * ``inverse_default``: name of an inverse method to set the default
+          value when the value is editable
+
+        Example::
+
+            @property
+            def _server_env_fields(self):
+                base_fields = super()._server_env_fields
+                sftp_fields = {
+                    "sftp_server": {
+                        "getter": "get",
+                    },
+                    "sftp_port": {
+                        "getter": "getint",
+                    },
+                    "sftp_login": {},
+                    "sftp_password": {},
+                }
+                sftp_fields.update(base_fields)
+                return sftp_fields
+        """
+        return {}
+
+    @api.model
+    def _server_env_global_section_name(self):
+        """Name of the global section in the configuration files
+
+        Can be customized in your model
+        """
+        return self._name.replace(".", "_")
+
+    @api.multi
+    def _server_env_section_name(self):
+        """Name of the section in the configuration files
+
+        Can be customized in your model
+        """
+        self.ensure_one()
+        base = self._server_env_global_section_name()
+        return ".".join((base, self.name))
+
+    @api.multi
+    def _server_env_read_from_config(self, field_name, config_getter):
+        self.ensure_one()
+        global_section_name = self._server_env_global_section_name()
+        section_name = self._server_env_section_name()
+        try:
+            # at this point we should have checked that we have a key with
+            # _server_env_has_key_defined so we are sure that the value is
+            # either in the global or the record config
+            getter = getattr(serv_config, config_getter)
+            if (section_name in serv_config
+                    and field_name in serv_config[section_name]):
+                value = getter(section_name, field_name)
+            else:
+                value = getter(global_section_name, field_name)
+        except Exception:
+            _logger.exception(
+                "error trying to read field %s in section %s",
+                field_name,
+                section_name,
+            )
+            return False
+        return value
+
+    @api.multi
+    def _server_env_has_key_defined(self, field_name):
+        self.ensure_one()
+        global_section_name = self._server_env_global_section_name()
+        section_name = self._server_env_section_name()
+        has_global_config = (
+            global_section_name in serv_config
+            and field_name in serv_config[global_section_name]
```

Comment:
`field_name in serv_config.get(global_section_name, {})`?

Reply:
ah no I already tried that, but it's not a dict, you get a `configparser.NoSectionError` exception when using `get` and the section does not exists

Reply:
:open_mouth: :+1: